x/wasm-storage/keeper/msg_server.go

```diff
@@ -173,3 +173,16 @@ func unzipWasm(wasm []byte) ([]byte, error) {
 	}
 	return unzipped, nil
 }
+
+func (m msgServer) UpdateParams(goCtx context.Context, req *types.MsgUpdateParams) (*types.MsgUpdateParamsResponse, error) {
+	if m.GetAuthority() != req.Authority {
+		return nil, fmt.Errorf("invalid authority; expected %s, got %s", m.GetAuthority(), req.Authority)
+	}
+
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	if err := m.SetParams(ctx, req.Params); err != nil {
```

Comment:
validate params before setting?

Reply:
The msg (`MsgUpdateParams`) would've been validated in `Validate()`:
```golang
func (m *MsgUpdateParams) Validate() error {
	if _, err := sdk.AccAddressFromBech32(m.Authority); err != nil {
		return fmt.Errorf("invalid authority address: %s", err)
	}

	if err := m.Params.ValidateBasic(); err != nil {
		return err
	}

	return nil
}
```